server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -219,14 +223,37 @@ public CachingClusteredClient(
     final List<TimelineObjectHolder<String, ServerSelector>> filteredServersLookup =
         toolChest.filterSegments(query, serversLookup);
 
+    // Absent Optional indicate a null rangeset which represents range cannot be determined
+    final Map<String, Optional<RangeSet<String>>> dimensionRangeMap = Maps.newHashMap();
+    final DimFilter filter = query.getFilter();
+
+    // Filter unneeded chunks based on partition dimension
```

Comment:
Could you extract this logic into a utility function?

I'm asking because there are folks working on querying historical nodes or druid segments directly (#2330, #2880, https://groups.google.com/d/msg/druid-development/FK5D162ao74/wRTNWAzsCQAJ, https://github.com/SparklineData/spark-druid-olap) and this code would be useful to those efforts.